Add `siteIsAtomic` property to remote logging error data

chihsuan opened the pull request:
### Changes proposed in this Pull Request:

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

Closes https://github.com/Automattic/wc-calypso-bridge/issues/1537

This PR introduces a `siteIsAtomic` property to the remote logging data in both PHP and JavaScript, indicating whether the site is Atomic. This addition will help in effectively monitoring and analyzing errors during release "bakes" on Atomic sites.

**Why using `siteIsAtomic` as a property name?**

I checked the logstash and found that existing logs have `isAtomic` and `siteIsAtomic` properties fields. However, only `siteIsAtomic` has relevant logs, so I decided to use the same field name here.

**Do we need to check if the site is Atomic?**

If I understand correctly, the `wc-calypso-bridge` plugin is loaded in wpcomsh so it should only be loaded on Atomic sites. But to make sure, I still add the `IS_ATOMIC` constant check.

<!-- The next section is mandatory. If your PR doesn't require testing, please indicate that you are purposefully omitting instructions. -->

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Otherwise, please include detailed instructions on how these changes can be tested. Please, make sure to review and follow the guide for writing high-quality testing instructions below. -->

- [ ] Have you followed the [Writing high-quality testing instructions guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions)?

1. Create an E-commerce site on WordPress.com
2. Upload and activate the **local** `wc-calypso-bridge` plugin
3. Upload and activate the beta tester plugin https://github.com/woocommerce/woocommerce/releases/download/wc-beta-tester-3.0.0/woocommerce-beta-tester.zip
4. Go to `/wp-admin/tools.php?page=woocommerce-admin-test-helper&tab=remote-logging`
5. Enable remote logging
6. Click on `Simulate Core Exception` to simulate a PHP error and fresh the page to trigger the error
7. Check the logs in logstash PCYsg-11XN-p2
8. Confirm that the logs have `siteIsAtomic = true`.
9. Enable debug logging `window.localStorage.setItem( 'debug', 'wc:remote-logging');`
10. Click on `Simulate Core Exception` to simulate a JavaScript error
11. Go to any WC Admin page
12. Check the logs in console/network tab first to see if the error is sent. When I tested, the API returned an error because I reached the API limit.
13. If it's sent, check the logs in logstash to confirm that the logs have `siteIsAtomic = true`. If it's not sent due to the API limit, simply check the debug logs in the console to verify the error data contains `siteIsAtomic = true`.


<img width="924" alt="Screenshot 2025-03-04 at 16 37 14" src="https://github.com/user-attachments/assets/c4bb7503-ac2b-4ac4-98a7-6cffc7d80aa8" />

![Screenshot 2025-03-04 at 17 27 45](https://github.com/user-attachments/assets/47987125-3c91-439c-b8f2-c5e570b7e5d1)


<img width="946" alt="Screenshot 2025-03-04 at 16 23 35" src="https://github.com/user-attachments/assets/6e8303ae-b6d3-4ccd-b026-78a4428f99f1" />



<!-- End testing instructions -->

### Other information:

-   [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?

<!-- Mark completed items with an [x] -->

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

chihsuan committed:
Add Atomic site detection and logging support

- Add `isAtomic` flag to shared script data
- Implement remote logging filter for Atomic sites in JavaScript
- Add method to include Atomic site property in WooCommerce logs
- Update version placeholder in main class file

chihsuan committed:
Add changelog

ilyasfoo reviewed (commented):
Tests well with a few comments!

#### Review checklist

| Parameter  | Result |
| ------------- | ------------- |
| Approach  | ✅ All good |
| Implementation  | ✅ All good |
| Automated tests  | Not applicable |
| Naming  | ✅ All good |
| Typing  | 👍  Almost good (left a comment) |
| Code comments  | ✅ All good |
| Changelog  | ✅ All good |
| Documentation  | Not applicable |
| PR description  | ✅ All good |
| Testing instructions  | ✅ All good |
| Manual testing  | ✅ Works as expected |

#### Scenarios tested

| Scenario  | Result | Screenshot |
| ------------- | ------------- | ------------- |
| Simulated error in PHP  | ✅ Functionality worked correctly |  |
| Simulated error in Javascript  | ✅ Functionality worked correctly |  |

ilyasfoo commented on class-wc-calypso-bridge.php:
```diff
@@ -318,6 +322,23 @@ public function log_message( $message, $level = 'debug', $context = 'dotcom-ecom
+		if ( ! isset( $log_data['properties'] ) ) {
+			$log_data['properties'] = array();
+		}
+		$log_data['properties']['siteIsAtomic'] = true;
```
<img width="214" alt="image" src="https://github.com/user-attachments/assets/7d8c69fc-11d6-4f2c-885c-90f10f70dcc4" />

chihsuan replied:
@ilyasfoo I agree that `siteIsAtomic` isn't an ideal name. However, it's already in use by WPCOM. (fbhepr%2Sfrnepu%3Scebwrpg%3Qjcpbz%26shyy%3QfvgrVfNgbzvp%26qrsf%3Q%26ersf%3Q%26cngu%3Q%26uvfg%3Q%26glcr%3Q%26keq%3Q%26aa%3Q1-og) as mentioned in the PR description. If we switch to `site_is_atomic`, we'll end up with two properties in the log: `siteIsAtomic` and `site_is_atomic`, which could be confusing. Let me know if you have any suggestions.

ilyasfoo replied:
I got you, sorry I missed that in the PR description! I agree staying with `siteIsAtomic` is the right way forward 👍

adrianduffell reviewed (approved):
Nice! It's LGTM pending Ilyas' suggestions 🚀

chihsuan committed:
Merge branch 'master' into add/remote-logging-filter-atomic

chihsuan committed:
Address feedback

chihsuan commented on class-wc-calypso-bridge.php:
```diff
@@ -318,6 +322,23 @@ public function log_message( $message, $level = 'debug', $context = 'dotcom-ecom
+	 * @return array Modified log data with Atomic site information.
+	 */
+	public function add_site_atomic_property_to_logs( $log_data ) {
+		if ( ! isset( $log_data['properties'] ) ) {
```
Good point! Updated the check in 9d03a027b4cf353a4278954abef0330f77758d60.

ilyasfoo commented on class-wc-calypso-bridge.php:
```diff
@@ -318,6 +322,23 @@ public function log_message( $message, $level = 'debug', $context = 'dotcom-ecom
+	 * @return array Modified log data with Atomic site information.
+	 */
+	public function add_site_atomic_property_to_logs( $log_data ) {
+		if ( ! is_array( $log_data ) || ! isset( $log_data['properties'] ) ) {
+			$log_data['properties'] = array();
+		}
```
I think this is still not 100%, ChatGPT generated this fix:

```php
if ( ! is_array( $log_data ) ) {
    $log_data = array();
}

if ( ! isset( $log_data['properties'] ) ) {
    $log_data['properties'] = array();
}
```

chihsuan replied:
Thanks, updated in df528f90ed99165763c0f19a4dc79121de16611c.

ilyasfoo reviewed (changes requested):
@chihsuan One more comment 🙏

chihsuan committed:
Improve log data handling

- Ensure log data is always an array
- Initialize properties array if not present
- Simplify logic for adding Atomic site property to logs

chihsuan committed:
Merge branch 'master' into add/remote-logging-filter-atomic

ilyasfoo reviewed (approved):
Awesome, thanks for addressing feedback, @chihsuan! LGTM 🚢
